service/src/main/java/org/jboss/sbomer/service/feature/sbom/features/umb/consumer/ErrataNotificationHandler.java

```diff
@@ -36,43 +44,105 @@
 
 @ApplicationScoped
 @Slf4j
-@Setter
 public class ErrataNotificationHandler {
 
     @Inject
     @RestClient
+    @Setter
     ErrataClient errataClient;
 
+    @Inject
+    @Setter
+    FeatureFlags featureFlags;
+
     public void handle(String message) throws JsonProcessingException {
 
         ErrataStatusChangeMessageBody errataStatusChange = ErrataMessageHelper.fromStatusChangeMessage(message);
-        log.info("Fetching Errata information for erratum with id {}", errataStatusChange.getErrataId());
+        log.info("Fetching Errata information for erratum with id {}...", errataStatusChange.getErrataId());
 
-        Errata erratum = errataClient.getErratum(String.valueOf(errataStatusChange.getErrataId()));
-        log.info("Fetched erratum \n{}", erratum);
-
-        if (erratum.getDetails().isEmpty()) {
-            log.info("Mmmmm I don't know how to get the release information, group_id is empty...");
+        if (!featureFlags.errataApiRead()) {
+            log.warn("Errata API integration is disabled, the UMB message won't be used!!");
             return;
         }
 
-        ErrataRelease erratumRelease = errataClient
-                .getReleases(String.valueOf(erratum.getDetails().get().getGroupId()));
-        log.info("Fetched erratum release {}", erratumRelease);
-        log.info(
-                "** This advisory is related to Product '{}' and ProductRelease '{}'. We don't know how to retrieve the ProductVariant yet.**",
-                erratumRelease.getData().getRelationships().getProduct().getShortName(),
-                erratumRelease.getData().getAttributes().getName());
+        String summary = retrieveAllErratumData(errataStatusChange.getErrataId());
+        System.out.println(summary);
+    }
+
+    private String retrieveAllErratumData(Long errataId) {
```

Comment:
This is a temporary method, right? I assume so and I skipped reviewing it.

Reply:
yes sure it's a temp one to print all the info retrieved!